src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -89,6 +89,70 @@ internal WebApplicationBuilder(WebApplicationOptions options, Action<IHostBuilde
         WebHost = new ConfigureWebHostBuilder(webHostContext, Configuration, Services);
     }
 
+    internal WebApplicationBuilder(WebApplicationOptions options, bool slim, Action<IHostBuilder>? configureDefaults = null)
+    {
+        Debug.Assert(slim, "should only be called with slim: true");
+
+        var configuration = new ConfigurationManager();
+
+        // TODO: Remove when DI no longer has RequiresDynamicCodeAttribute https://github.com/dotnet/runtime/pull/79425
+#pragma warning disable IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
+        _hostApplicationBuilder = new HostApplicationBuilder(new HostApplicationBuilderSettings
+        {
+            Args = options.Args,
+            ApplicationName = options.ApplicationName,
+            EnvironmentName = options.EnvironmentName,
+            ContentRootPath = options.ContentRootPath,
+            Configuration = configuration,
```

Comment:
I know we're planning on adding new trimmable API for this, but would it make sense to use `DisableDefaults = true` in the meantime?

Reply:
Yeah, I think that makes sense. It will have the same behavior, without the size reduction (yet).

Reply:
I tried this change, and it looks like it broke a bunch of tests.

Should I change it back to allow the tests to pass for now? Or should I fix the tests for this case?

Reply:
Thanks for fixing the tests!